cluster_tools/tests/test_slurm.py

```diff
@@ -284,11 +291,70 @@ def test_slurm_max_array_size() -> None:
 
             assert all(array_size <= max_array_size for array_size in occurences)
     finally:
-        chcall(f"sed -i 's/{command}//g' /etc/slurm/slurm.conf && scontrol reconfigure")
+        chcall(
+            f"sed -ci 's/{command}//g' /etc/slurm/slurm.conf && scontrol reconfigure"
+        )
         reset_max_array_size = executor.get_max_array_size()
         assert reset_max_array_size == original_max_array_size
 
 
+# TODO: Comment back in after the test ran through in the CI
+# @pytest.mark.skip(
+#     reason="This test takes more than a minute and is disabled by default. Execute it when modifying the RemoteTimeLimitException code."
+# )
+def test_slurm_time_limit() -> None:
+    # Time limit resolution is 1 minute, so request 1 minute
+    executor = cluster_tools.get_executor(
+        "slurm", debug=True, job_resources={"time": "0-00:01:00"}
+    )
+
+    with executor:
+        # Schedule a job that runs for more than 1 minute
+        futures = executor.map_to_futures(sleep, [80])
+        concurrent.futures.wait(futures)
+
+        # Job should have been killed with a RemoteTimeLimitException
+        assert all(
+            isinstance(fut.exception(), cluster_tools.RemoteTimeLimitException)
+            for fut in futures
+        )
+
+
+def test_slurm_memory_limit() -> None:
+    # Request 1 MB
+    executor = cluster_tools.get_executor(
+        "slurm", debug=True, job_resources={"mem": "1M"}
+    )
+
+    original_gather_frequency_config = "JobAcctGatherFrequency=30"  # from slurm.conf
+    new_gather_frequency_config = "JobAcctGatherFrequency=1"
+
+    try:
+        # Increase the frequency at which slurm checks whether a job uses too much memory
+        chcall(
+            f"sed -ci 's/{original_gather_frequency_config}/{new_gather_frequency_config}/g' /etc/slurm/slurm.conf && scontrol reconfigure"
+        )
```

Comment:
how about extracting this into a helper function, since it's also used in line 294?

Reply:
if you want to get extra fancy, the helper function could return an `undo` function. something like this:

```python
def change_config(old, new):
  chcall(f"sed {old} ... {new}")
  def undo():
    chcall(f"sed {new} ... {old}")
  return undo
```